src/Docker/Runner/StateFile.php

```diff
@@ -122,17 +128,30 @@ public function persistState()
             $configuration->setComponentId($this->componentId);
             $configuration->setConfigurationId($this->configurationId);
             try {
+                $encryptedStateData = $this->encryptorFactory->getEncryptor()->encrypt($this->currentState, ProjectWrapper::class);
                 if ($this->configurationRowId) {
+                    $storedState = $components->getConfigurationRow($this->componentId, $this->configurationId, $this->configurationRowId)['state'];
+
+                    if (!isset($storedState[self::NAMESPACE_PREFIX])) {
```

Comment:
ta podminka je podle me blbost, kdyz bude `state->im->something` tak to prepise na `state->component->[]` podle me by tu melo byt jen ` $storedState[self::NAMESPACE_PREFIX] = $encryptedStateData;` a nic vic

Reply:
takhle to určitě bude, ale až po té migraci! teď potřebujeme chvíli detekovat, jestli ten state je `legacy` nebo `namespaced` a umět mezi nima jednoznačně vybrat. až se všechny state zmigrujou (https://github.com/keboola/connection/issues/1854), tak další PR na docker-bundle bude vymazávat podmínku a bude slepě zapisovat `$storedState[self::NAMESPACE_PREFIX] = $encryptedStateData;`

Reply:
no tak jestli tohle predpoklada, ze ve state neni nic jinyho state, tak vysledkem bude vzdycky vnorenej state (coz je v poradku), ale timpadem ta podminka vubec nic nedela

Reply:
```php
        if (isset($state[self::NAMESPACE_PREFIX])) {
            $this->state = $state[self::NAMESPACE_PREFIX];
        } else {
            $this->state = $state;
        }
```

jasně, že dělá, do `state.json` načte stejný state ať je uloženej jako `legacy` nebo `namespaced`

Reply:
no to je ale jina podminka, ta je v poradku

Reply:
kuafix, no jasně! už to vidím, asi jsem měl v tu chvíli nějakej zkrat a myslel jsem si, že nejde zapisovat do neinicializovaných klíčů pole stejně jako z nich nejde číst, hahaha!

Reply:
no ale moment! ono to tam fakt má bejt, jenom to není zřejmý. pokud je state `legacy`, musí se celej přepsat.

**configuration**
```
{
  "state": {
    "key": value"
  }
}
```
**state.json**
```
{
  "key": value"
}
```

**:x: result**
`$storedState[self::NAMESPACE_PREFIX] = $encryptedStateData;`
```
{
  "state": {
    "key": value",
    "component": {
      "key": "value"
    }
  }
}
```

**:white_check_mark: result**
`$storedState = [self::NAMESPACE_PREFIX => $encryptedStateData];`
```
{
  "state": {
    "component": {
      "key": "value"
    }
  }
}
```

Reply:
aha, protoze se tam dava ten stored state, hm a proc tam teda vlastne neni jen

Reply:
Jasně, přemejšlel jsem moc dopředu. Takhle to bude nejjednodušší a nemusí se `$storedState` teď vůbec načítat. To se vyřeší až v dalším pull requestu. To načtení a zapsání state se mi moc nelíbí, o to hůř, že se to bude zřejmě dít dvakrát za chodu komponenty (component state, output mapping state) a ani jedno z toho není v transakci nebo verzovaný, aby se předešlo kolizím. Na to by bylo nejlepší umět asi updatovat část state přímo na API.

Reply:
protoze je to takovej kockopes mezi tim co je potreba ted a co bude potreba potom?

Reply:
aha, nez jsem na to dosel, tak uz jsi na to dosel taky :)

Reply:
no afaik by se to melo zapsat vsechno naraz, protoze state se zapisuje az nakonci vseho pokud nedojde k chybe

Reply:
jo, jasně, tak se to aspoň bude dít jenom jednou, ale stejně se mi to moc nelíbí :-(